### ISSUE TITLE
Catalog Sorting doesn't display on a Home page

### ISSUE THREAD

**Issue by kmanijak** [woothemes/woocommerce#42387]
### Steps to reproduce
1. Go to Editor
2. Edit Home template
3. Add _Catalog Sorting_ and _Products_ blocks - they both show up in the Editor
4. Save and go to the frontend home page of your shop

### Expected
Catalog Sorting is displayed on top of the Products block

### Actual
Catalog Sorting is not displayed.

### Screenshots
| Editor | Frontend |
| --- | --- |
| <img width="1343" alt="image" src="https://user-images.githubusercontent.com/20098064/235090238-adff0997-e015-462e-bd1b-0970f7fad203.png"> | <img width="1227" alt="image" src="https://user-images.githubusercontent.com/20098064/235090270-a91f1491-8719-4032-85da-bbe884d7b3b3.png"> |

### Additional notes
Catalog Sorting block is a wrapper around the classic widget, which is not available to be added on the Home page. That means one of the temporary solution could be to allow using Catalog Sorting only on a product archive templates (it's not definitive, just adding more context).

**Comment by gigitux** [comment]
@tjcafferkey This issue presents an opportunity to improve the `registerProductBlockType` function by making it more generic and allowing the list of templates where the block should be available to be passed as an argument.  

Currently, if not mistaken, `registerProductBlockType` registers the block on all WooCommerce templates.

**Comment by tjcafferkey** [comment]
>@tjcafferkey This issue presents an opportunity to improve the registerProductBlockType function by making it more generic and allowing the list of templates where the block should be available to be passed as an argument.

I’d argue there’s real value in keeping its current, focused purpose intact. It’s designed specifically to register blocks that rely on product context (i.e., Single Product template or Single Product block), and this constraint ensures consistency and predictability for us. Broadening its scope too much could dilute that clarity, potentially leading to misuse or unexpected behavior in templates where product context isn’t guaranteed.

If we wanted to explore more flexibility, a separate API might be a cleaner solution. Something purpose-built to handle per-template registration without muddying the intent of `registerProductBlockType`.

I'm keen to know your thoughts on this though!

>Currently, if not mistaken, registerProductBlockType registers the block on all WooCommerce templates.

Not quite, it’s limited to the Single Product template or Single Product block, which aligns with its product-context focus.

**Comment by gigitux** [comment]
> If we wanted to explore more flexibility, a separate API might be a cleaner solution. Something purpose-built to handle per-template registration without muddying the intent of `registerProductBlockType`.

You’re right! We could create a separate API and potentially use it `registerProductBlockType` function.

**Comment by tjcafferkey** [comment]
>You’re right! We could create a separate API and potentially use it registerProductBlockType function.

I'm glad we're on the same page, thank you! I think this could be a great cooldown task potentially! Do you think it's worth opening a separate issue (linking this) with more intentional scope for the new internal API?

**Comment by gigitux** [comment]
> > You’re right! We could create a separate API and potentially use it registerProductBlockType function.
> 
> I'm glad we're on the same page, thank you! I think this could be a great cooldown task potentially! Do you think it's worth opening a separate issue (linking this) with more intentional scope for the new internal API?

It makes sense! Do you have the bandwidth to create it and link it to this issue?

**Comment by tjcafferkey** [comment]
Of course ~Grok does~ I do.
